modulemd/modulemd-modulemetadata.c

```diff
@@ -435,11 +409,12 @@ modulemd_modulemetadata_set_buildrequires (ModulemdModuleMetadata *self,
                                            GHashTable             *buildrequires)
 {
   g_return_if_fail (MODULEMD_IS_MODULEMETADATA (self));
+  g_return_if_fail (buildrequires);
 
-  if (buildrequires != self->buildrequires)
+  if (self->buildrequires != buildrequires)
     {
-      g_hash_table_destroy (self->buildrequires);
-      modulemd_copy_htable_strings (buildrequires, &self->buildrequires);
+      g_hash_table_unref (self->buildrequires);
+      self->buildrequires = g_hash_table_ref (buildrequires);
       g_object_notify_by_pspec (G_OBJECT (self),
```

Comment:
well, then we should create Object on top of GHashTable and make functions which wrap g_hash_table_*() functions and sending signals. Hard destory in finalize() and deepcopy everywhere is **worst thing ever**.